File: app/src/main/java/space/pal/sig/repository/FactRepository.java
package space.pal.sig.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import io.reactivex.Completable;
import lombok.RequiredArgsConstructor;
import space.pal.sig.model.Fact;
import space.pal.sig.model.database.FactDao;

/**
 * SpaceNow
 * Created by Catalin on 7/29/2020
 **/
@RequiredArgsConstructor
public class FactRepository {

    private final FactDao factDao;

    public Completable insertOrUpdate(List<Fact> factList) {
        return factDao.insert(factList.toArray(new Fact[0]));
    }

    public Completable insertOrUpdate(Fact fact) {
        return factDao.insert(fact);
    }

    public LiveData<List<Fact>> findAll() {
        return factDao.findAll();
    }

}

File: app/src/main/java/space/pal/sig/repository/FactRepository.kt
package space.pal.sig.repository

import androidx.lifecycle.LiveData
import space.pal.sig.database.dao.FactDao
import space.pal.sig.model.entity.Fact

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
class FactRepository(
        private val factDao: FactDao
) {

    fun save(fact: Fact) {
        return factDao.save(fact)
    }

    fun findAll(): LiveData<MutableList<Fact>> {
        return factDao.findAll()
    }

}